src/libsystemd/sd-bus/bus-introspect.c

```diff
@@ -101,6 +101,12 @@ static int introspect_write_arguments(struct introspect *i, const char *signatur
 
                 fprintf(i->f, "   <arg type=\"%.*s\"", (int) l, signature);
 
+                if(*names) {
+                    int ret = fprintf(i->f, " name=\"%s\"", names)-8;
+                    if(ret>0)
+                        names += ret+1;
```

Comment:
I think using strlen() here is a lot nicer. The compiler is pretty good add optimizing this away, and it's nicer to read.